### ISSUE TITLE
Error when opening web player

### ISSUE THREAD

**Issue by Keavon** [tarkah/karaoke-rs#9]
With `use_web_player: true` set and upon clicking the "Player" tab in the web app, the console spits out:

```
[2020-01-27T02:49:37Z INFO  karaoke_rs::websocket] Connection from 127.0.0.1:64870
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: IoError(Os { code: 10035, kind: WouldBlock, message: "A non-blocking socket operation could not be completed immediately." })', src\libcore\result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', src\libcore\result.rs:1165:5
```

I am using Windows 10 64-bit and the `karaoke-rs-v0.8.0-x86_64-pc-windows-msvc.zip` build.

**Comment by tarkah** [comment]
Ahh, I never tested this on Windows :P appears to be an issue with the websocket server used to communicate with the web player. I'll dig into this week, I have a Windows machine to test on.

**Comment by tarkah** [comment]
Easy fix! Windows returns an error when trying to receive a message on a non-blocking socket when no message has since been received. I was able to safely ignore these errors so no panic would occur. This has been merged.

**Comment by tarkah** [comment]
@Keavon I'm going to try to knock out all your feature requests before I push out a new release. Are you comfortable building from source in the meantime, or do you want me to push out a 0.8.1 release with this patch so you can use?

**Comment by tarkah** [comment]
Nvm, I'll just push out 0.8.1 w/ this fix and remove the windows binary from the 0.8.0 release so other people won't see this issue.

**Comment by Keavon** [comment]
I can probably build from source. I've been learning Rust recently but only brushed the surface so far, but it looks like Cargo makes it nice and easy compared to the disaster that is C in setting up a build environment. However I can probably wait a week or two anyways since I don't need these features immediately (the user-templated file format would be most useful in the short-term). I'll also throw a few more feature requests your way once you knock out most of these (mainly, server-side searching for large libraries).
